app/src/main/java/net/squanchy/schedule/tracksfilter/ScheduleTracksFilter.kt

```diff
@@ -56,97 +63,133 @@ class ScheduleTracksFilterActivity : AppCompatActivity() {
             setDrawable(resources.getDrawable(R.drawable.filters_separator, theme))
             setOrientation(FlexboxItemDecoration.BOTH)
         })
-        trackFiltersList.adapter = trackAdapter
         trackFiltersList.itemAnimator = null
     }
 
     private fun Set<Track>.addOrRemove(track: Track, selected: Boolean): Set<Track> =
         if (selected) this + track else this - track
 
+    private fun animateDisappearance() {
+        backgroundDim.isEnabled = false
+
+        val centerX = closeButton.x + closeButton.width / 2
+        val centerY = closeButton.y + closeButton.height / 2
+
+        ViewAnimationUtils.createCircularReveal(
+            filtersRoot,
+            centerX.toInt(),
+            centerY.toInt(),
+            hypot(filtersRoot.width.toDouble(), filtersRoot.height.toDouble()).toFloat(),
+            0F
+        ).apply {
+            duration = resources.getInteger(R.integer.track_filters_disappear_duration).toLong()
+
+            doOnEnd {
+                filtersRoot.visibility = View.INVISIBLE
+                finish()
+            }
+
+            start()
```

Comment:
shouldn't this done outside of `apply`? I know it doesn't change the behavior but technically you don't do it till the object is ready

Reply:
I don't get what is the actual difference, even just semantically?

Reply:
semantically indeed, apply feels like something you'd call because you don't have a builder, and should return the object ready to be used with `start`

Reply:
/shrug can do, doesn't really make any difference to me

Reply:
609e9a69c3e622999dd949801436f84ded7d768e